### PR TITLE
Update to tabled layout v2

### PR DESCRIPTION
**Pull request by cassiogreco**
This is a PR to update the UI of the comparison widget to a tabled format as in the image below.

screen shot 2017-01-10 at 16 44 04

Some styles due to differences in bootstrap are a bit different. 

This design comes from a few different tests as can be seen:
https://docs.google.com/presentation/d/14WX0WcjuPfuCd8G9t0hpMWWGcoldDRJBUJPmVVqT8lE/edit#slide=id.g15655fd308_0_5 (the latest with statistically significant improvement in conversion)

And:
https://docs.google.com/document/d/1jaD2qN7MR1-8qx8LbPlPLp3CeEfT4QAueSJm2gJru4Q/edit
https://docs.google.com/document/d/1AjsYigQXUEhAspC-6_YENR79wol17pSWSEM6Jg5bvfk/edit

### CODE REVIEW

**Comment by oliverviljamaa** [comment]
I think we should expose both the new component and the old one for the time being – as the new one is highly likely going to be used, I'd name this `ComparisonTable` and the old one something else, `ComparisonBarTable` or something (better, hopefully).
This would give us the ability to A/B test it against the older version 😈

File: src/comparison-table/Provider.js [hunk 1]
```diff
@@ -10,59 +10,39 @@ export default class Provider extends Component {
...
 
-            <Amount className="h2 font-weight-normal text-no-wrap m-l-2" style={{verticalAlign: 'middle', margin: 0}} currency={this.props.source} value={this.props.fees + this.props.hiddenFees}/>
-          </div>
-          <div>
-            <small className="text-primary">Fee</small>
-            {this.props.hiddenFees > 0 ? <small className="text-danger m-l-2">Rate markup</small> : null}
-          </div>
+         <td className="hidden-md hidden-lg hidden-xl text-xs-center">
+            {this.hasNoHiddenFees() ?
+              <Amount className="h5 text-success font-weight-bold p-t-1" style={{verticalAlign: 'middle'}} value={this.props.rate} precision={4}></Amount> :
```
Comment (kdelemme) [inline src/comparison-table/Provider.js]: we pass `verticalAlign: 'middle'` all the time. Until we have different possible values, does it make sense to have this?

Comment (cassiogreco) [inline src/comparison-table/Provider.js]: need to clean those

File: demo/comparison-table/ComparisonTableDemo.js [hunk 2]
```diff
@@ -40,8 +40,13 @@ export default class Demo extends Component {
           source={'GBP'}
           target={'EUR'}
           amount={this.state.amount}
-          providers={['transferwise', 'lloyds']}
         />
```
Comment (oliverviljamaa) [inline demo/comparison-table/ComparisonTableDemo.js]: Can we update the demo page to have them separately on the page? Separate title, test amounts, etc.?

File: src/comparison-bar-table/ComparisonTable.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
```
Comment (oliverviljamaa) [inline src/comparison-bar-table/ComparisonTable.js]: Maybe worth changing the name as well?

File: src/comparison-table/Amount.js [hunk 4]
```diff
@@ -4,7 +4,7 @@ export default class Amount extends Component {
   render() {
     const {currency, value, ...props} = this.props;
     return (
-      <span {...props}>{parseFloat(value).toFixed(2)} <span className="small">{currency}</span></span>
+      <strong {...props}>{parseFloat(value).toFixed(this.props.precision || 2)} <bold className="text-primary font-weight-bold">{currency}</bold></strong>
```
Comment (oliverviljamaa) [inline src/comparison-table/Amount.js]: Why `strong` AND `bold` AND `.font-weight-bold`?

File: src/comparison-table/ComparisonTable.js [hunk 5]
```diff
@@ -100,18 +100,21 @@ export default class ComparisonTable extends Component {
...
-        <table className="table">
-          <tbody>
-            <tr>
-              <th>Provider</th>
-              <th className="text-xs-right hidden-xs text-md-left text-xs-nowrap">Total cost of sending {this.props.amount} {this.props.source} to {this.props.target}</th>
-              <th className="text-xs-right">Amount received</th>
+      <div className="table-responsive">
+        <table className="table table-bordered">
+          <thead>
+            <tr className="bg-primary text-inverse">
```
Comment (oliverviljamaa) [inline src/comparison-table/ComparisonTable.js]: Is the rounded corner rounded now?

File: src/comparison-table/ComparisonTable.less [hunk 6]
```diff
@@ -1,61 +1,40 @@
...
+
+    @media (min-width: @screen-sm-min) {
+      background-color: #fff;
+    }
+  }
+
+  .provider-logo {
+    max-width: 6em;
+    max-height: 2em;
+    padding-left: 8px;
```
Comment (oliverviljamaa) [inline src/comparison-table/ComparisonTable.less]: We shouldn't need this, we should just style the cell padding in `td`, no?

File: demo/comparison-table/ComparisonTableDemo.js (hunk 2 above)
Comment (cassiogreco) [inline demo/comparison-table/ComparisonTableDemo.js]: I can see it having a separate title but why different test amounts?

File: src/comparison-table/ComparisonTable.js (hunk 5 above)
Comment (cassiogreco) [inline src/comparison-table/ComparisonTable.js]: no :/ the bg-primary is rectagular

File: demo/comparison-table/ComparisonTableDemo.js (hunk 2 above)
Comment (oliverviljamaa) [inline demo/comparison-table/ComparisonTableDemo.js]: When you're testing one, it's likely you don't want the other one updating and making the call, is how I see it.

File: src/comparison-bar-table/ComparisonTable.js (hunk 3 above)
Comment (cassiogreco) [inline src/comparison-bar-table/ComparisonTable.js]: of the import?

File: src/comparison-table/ComparisonTable.js (hunk 5 above)
Comment (oliverviljamaa) [inline src/comparison-table/ComparisonTable.js]: Let's look into fixing that.

File: demo/comparison-table/ComparisonTableDemo.js (hunk 2 above)
Comment (cassiogreco) [inline demo/comparison-table/ComparisonTableDemo.js]: yeah i was playing around and its better to have them separate. changed :)

File: src/comparison-bar-table/ComparisonApi.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+const API_URL = 'https://api.transferwise.com/v1';
```
Comment (oliverviljamaa) [inline src/comparison-bar-table/ComparisonApi.js]: Can we move this to `src`?

File: src/comparison-bar-table/Disclaimer.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
```
Comment (oliverviljamaa) [inline src/comparison-bar-table/Disclaimer.js]: Could we share the disclaimer component as well?

File: src/comparison-table/Amount.js [hunk 9]
```diff
@@ -4,7 +4,7 @@ export default class Amount extends Component {
   render() {
     const {currency, value, ...props} = this.props;
     return (
-      <span {...props}>{parseFloat(value).toFixed(2)} <span className="small">{currency}</span></span>
+      <strong {...props}>{parseFloat(value).toFixed(this.props.precision || 2)} <strong className="text-primary">{currency}</strong></strong>
```
Comment (oliverviljamaa) [inline src/comparison-table/Amount.js]: Double `strong`.

File: src/comparison-table/ComparisonTable.js [hunk 10]
```diff
@@ -100,18 +100,21 @@ export default class ComparisonTable extends Component {
...
+              <th className="font-weight-normal">Provider</th>
```
Comment (oliverviljamaa) [inline src/comparison-table/ComparisonTable.js]: `.font-weight-normal`

File: src/comparison-table/ComparisonTable.less [hunk 11]
```diff
@@ -1,61 +1,40 @@
...
-    background-color: #f2f5f7;
+  .highest-amount {
+    font-size: 22px;
   }
-}
 
-@media (min-width: 544px) and (max-width: 767px) {
-  .tw-comparison-table .amount-received {
+  .amount-received {
     background-color: #f2f5f7;
```
Comment (oliverviljamaa) [inline src/comparison-table/ComparisonTable.less]: Could we import our variable for that as well?

File: src/comparison-table/Provider.js [hunk 12]
```diff
@@ -10,59 +10,39 @@ export default class Provider extends Component {
   }
 
   render() {
-    const barWidth = (((this.props.hiddenFees + this.props.fees) / this.props.maxFee) * 100) * 0.5;
-    const feeBarWidth = (this.props.fees / (this.props.hiddenFees + this.props.fees) * 100);
     return (
       <tr>
-        <td className="provider-name text-xs-nowrap">
-          <div className="h3 font-weight-normal m-t-3"><img className="provider-logo" src={this.props.logo} alt={this.props.name}/></div>
+        <td className="text-xs-nowrap">
```
Comment (oliverviljamaa) [inline src/comparison-table/Provider.js]: Does this need `.text-xs-nowrap`?

File: src/comparison-table/Provider.js [hunk 13]
```diff
@@ -10,59 +10,39 @@ export default class Provider extends Component {
...
-        <td className="hidden-xs hidden-sm">
-          <div className="m-t-3 text-xs-nowrap">
-            <div className="progress progress-lg" style={{position: 'relative', width: barWidth + '%', verticalAlign: 'middle', minWidth: '35px'}}>
-              {parseFloat(this.props.fees) > 0 ?
-                <div className="progress-bar" style={{width: feeBarWidth + '%'}}>
-                </div> : null}
-            </div>
+        <td className="hidden-md hidden-lg hidden-xl text-xs-right amount-received text-primary">
+          {this.props.highestAmount == this.props.amount ?
+            <Amount className="text-xs-nowrap" currency={this.props.target} value={this.props.amount}/> :
```
Comment (oliverviljamaa) [inline src/comparison-table/Provider.js]: Repetition?

File: src/comparison-table/Provider.js [hunk 14]
```diff
@@ -10,59 +10,39 @@ export default class Provider extends Component {
...
+         </td>
 
-            <Amount className="h2 font-weight-normal text-no-wrap m-l-2" style={{verticalAlign: 'middle', margin: 0}} currency={this.props.source} value={this.props.fees + this.props.hiddenFees}/>
-          </div>
-          <div>
-            <small className="text-primary">Fee</small>
-            {this.props.hiddenFees > 0 ? <small className="text-danger m-l-2">Rate markup</small> : null}
-          </div>
+         <td className="hidden-md hidden-lg hidden-xl text-xs-right">
+            {this.hasNoHiddenFees() ?
```
Comment (oliverviljamaa) [inline src/comparison-table/Provider.js]: Could we have this ternary in `className` instead?

File: src/comparison-table/Provider.js [hunk 15]
```diff
@@ -10,59 +10,39 @@ export default class Provider extends Component {
...
-          <div className="m-t-3">
-            <Amount className="m-b-0 h2 font-weight-normal text-xs-nowrap" currency={this.props.target} value={this.props.amount}/>
-            <small className="text-xs-nowrap">
-              {this.hasNoHiddenFees() ?
-                <div>Real rate <span className="text-default">{this.props.rate}</span></div> :
-                <div>{this.props.name} rate <span className="text-danger">{this.props.rate}</span></div>}
-            </small>
-          </div>
+        <td className="hidden-xs hidden-sm text-xs-right">
+            {this.hasNoHiddenFees() ?
```
Comment (oliverviljamaa) [inline src/comparison-table/Provider.js]: Ternary in `className`?

File: src/comparison-table/Provider.js [hunk 16]
```diff
@@ -10,59 +10,39 @@ export default class Provider extends Component {
...
+              <Amount className="text-success" value={this.props.rate} precision={4}></Amount> :
+              <Amount className="text-danger" value={this.props.rate} precision={4}></Amount>}
         </td>
+
+        <td className="hidden-xs hidden-sm text-xs-right">
+          {this.props.highestAmount == this.props.amount ?
+            <Amount className="highest-amount text-primary text-xs-nowrap amount-received" currency={this.props.target} value={this.props.amount}/> :
```
Comment (oliverviljamaa) [inline src/comparison-table/Provider.js]: Ternary in `className`?

**Pull request merged**